<?xml version="1.0" ?>

<launch>
  <arg name="prediction_time_horizon" default="10.0" />
  <arg name="prediction_sampling_delta_time" default="0.5" />
  <arg name="dist_threshold_for_searching_lanelet" default="3.0" />
  <arg name="delta_yaw_threshold_for_searching_lanelet" default="0.785" />
  <arg name="sigma_lateral_offset" default="0.5" /> <!-- [m]-->
  <arg name="sigma_yaw_angle" default="5.0" /> <!-- [angle degree] -->
  <arg name="vector_map_topic" default="/map/vector_map" />
  <arg name="output_topic" default="objects"/>
  <node pkg="map_based_prediction" exec="map_based_prediction" name="map_based_prediction" output="screen">
    <param name="perception_time_horizon" value="$(var prediction_time_horizon)" />
    <param name="prediction_sampling_delta_time" value="$(var prediction_sampling_delta_time)" />
    <param name="dist_threshold_for_searching_lanelet" value="$(var dist_threshold_for_searching_lanelet)" />
    <param name="delta_yaw_threshold_for_searching_lanelet" value="$(var delta_yaw_threshold_for_searching_lanelet)" />
    <param name="sigma_lateral_offset" value="$(var sigma_lateral_offset)" />
    <param name="sigma_yaw_angle" value="$(var sigma_yaw_angle)" />
    <remap from="/vector_map" to="$(var vector_map_topic)"/>
    <remap from="objects" to="$(var output_topic)"/>
  </node>
</launch>
